windows/OrientationWindows/OrientationWindows.cpp

```diff
@@ -0,0 +1,211 @@
+        if (mode == UserInteractionMode::Touch) {
+            if (m_displayInfo.AutoRotationPreferences() != DisplayOrientations::Landscape) {
+                DisplayInformation::AutoRotationPreferences(DisplayOrientations::Landscape);
+                LockDidChange("LANDSCAPE-LEFT");
+            }
+        }
+    }
+
+    void OrientationLockerModule::UnlockAllOrientations() noexcept {
+        UserInteractionMode mode = m_viewSettings.UserInteractionMode();
+        if (mode == UserInteractionMode::Touch) {
+            DisplayOrientations allOrientations = DisplayOrientations::Landscape | DisplayOrientations::LandscapeFlipped | DisplayOrientations::Portrait | DisplayOrientations::PortraitFlipped;
+            if (m_displayInfo.AutoRotationPreferences() != allOrientations) {
+                DisplayInformation::AutoRotationPreferences(allOrientations);
+                GetOrientation(LockDidChange);
+            }
+        }
+    }
+
+    void OrientationLockerModule::GetConstants(React::ReactConstantProvider& provider) noexcept {
+        if (!m_context.UIDispatcher().HasThreadAccess()) {
+            HANDLE ghSetConstantsEvent;
+            ghSetConstantsEvent = CreateEvent(
+                NULL,                       // default security attributes
+                TRUE,                       // manual-reset event
+                FALSE,                      // initial state is nonsignaled
+                TEXT("SetConstantsEvent")   // object name
+            );
+            if (ghSetConstantsEvent == NULL)
+            {
+                printf("CreateEvent for ghSetConstantsEvent failed (%d)\n", GetLastError());
+                return;
+            }
+            m_context.UIDispatcher().Post([&provider, ghSetConstantsEvent, this]() {
+                InitConstants();
+                provider.Add(L"initialOrientation", GetInitOrientation());
+                if (!SetEvent(ghSetConstantsEvent))
+                {
+                    printf("SetEvent for ghSetConstantsEvent failed (%d)\n", GetLastError());
+                    return;
+                }
+                });
+            DWORD dwWaitResult;
+            dwWaitResult = WaitForSingleObject(
+                ghSetConstantsEvent,
+                INFINITE);                  // wait period
+            switch (dwWaitResult)
+            {
+            case WAIT_OBJECT_0:
+                break;
+            default:
+                printf("Wait error for dwWaitResult (%d)\n", GetLastError());
+                break;
+            }
+            CloseHandle(ghSetConstantsEvent);
+        }
+        else {
+            InitConstants();
+            provider.Add(L"initialOrientation", GetInitOrientation());
+        }
+    }
+
+    std::string OrientationLockerModule::OrientationToString(DisplayOrientations orientation) noexcept {
+        std::string result = "";
+        if ((orientation & DisplayOrientations::Landscape) == DisplayOrientations::Landscape)
```

Comment:
can these values be somehow combined? why not just compare orientation == DisplayOrientations::Landscape, etc.